bindings/src/Capgemini.PowerApps.SpecFlowBindings/PowerAppsStepDefiner.cs

```diff
@@ -132,12 +127,34 @@ protected static ITestDriver TestDriver
         {
             get
             {
-                if (testConfig.BrowserOptions.BrowserType.SupportsProfiles())
+                if (!testConfig.BrowserOptions.BrowserType.SupportsProfiles())
+                {
+                    throw new NotSupportedException($"The {testConfig.BrowserOptions.BrowserType} does not support profiles.");
+                }
+
+                if (userProfileDirectories != null)
                 {
                     return userProfileDirectories;
                 }
 
-                throw new ArgumentException($"The {testConfig.BrowserOptions.BrowserType} does not support profiles.");
+                userProfileDirectories = new Dictionary<string, string>();
```

Comment:
I wouldn't assign `userProfileDirectories` until it's populated. Given that this is static and we'll be accessing from multiple threads it runs the risk of a race condition where you might get an empty dictionary returned because another thread has assigned it but hasn't finished populating it.

Reply:
Good point - I've updated to avoid this